src/controllers/scripting/controllerscriptenginebase.cpp

```diff
@@ -0,0 +1,251 @@
+#include "controllers/scripting/controllerscriptenginebase.h"
+
+#include "control/controlobject.h"
+#include "controllers/controller.h"
+#include "controllers/controllerdebug.h"
+#include "controllers/scripting/colormapperjsproxy.h"
+#include "errordialoghandler.h"
+#include "mixer/playermanager.h"
+
+ControllerScriptEngineBase::ControllerScriptEngineBase(Controller* controller)
+        : m_bDisplayingExceptionDialog(false),
+          m_pJSEngine(nullptr),
+          m_pController(controller),
+          m_bTesting(false) {
+    // Handle error dialog buttons
+    qRegisterMetaType<QMessageBox::StandardButton>("QMessageBox::StandardButton");
+
+    // Subclasses are responsible for adding paths to watch to m_scriptWatcher
+    // in their initializeScriptEngine method.
+    connect(&m_scriptWatcher,
+            &QFileSystemWatcher::fileChanged,
+            this,
+            &ControllerScriptEngineBase::reload);
+}
+
+ControllerScriptEngineBase::~ControllerScriptEngineBase() {
+    shutdown();
```

Comment:
You must not call a virtual function from a destructor, because it is missleading. It calls only ControllerScriptEngineBase::shutdown()

Reply:
fixed in 3383aef